addons/account/models/account_tax.py

```diff
@@ -653,6 +653,15 @@ def recompute_base(base_amount, fixed_amount, percent_amount, division_amount):
                         total_included_checkpoints[i] = base
                         store_included_tax_total = False
                 i -= 1
+            # In case we're computing the total_excluded from a value that includes the taxes, we should check
+            # if an early pay discount was applied on that value. If so, we apply the discount to the tax.
+            if self._context.get('early_pay_discount', False):
```

Comment:
You could clarify that the `early_pay_discount` must be `mixed` so it's clearer for posterity; `included` and `excluded` won't affect the calculation here